email_backend/src/email_backend.js

```diff
@@ -143,6 +147,40 @@ const make_email_backend = (templates) => {
         );
         if (mail_sent_successfully){
           response.send("200");
+
+          //Sending info to MongoDB server
+          const mongoose = require('mongoose');
```

Comment:
From what I've just read: "Require Throws error at runtime and Import throws error while parsing. Require is Nonlexical and Import is Lexical." I can see why Import might be better to use in our case, but I don't personally know too much about the history of JS with them 😅

Reply:
Those are the current, practical, differences between the two.

The JS specification (ECMAScript) didn't have modules for a long time. There were multiple community standards/libraries for adding modules to JavaScript over the years. CommonJS was an early one (used `require` syntax, I think it was adopted and mainly used as Node.js' default), then there was a spec called AMD that was implemented as RequireJS (also used `require`, was more browser focused than CommonJS). Finally, in ECMAScript 6 (also called ES6), they added an official module syntax to the JS spec using the keyword `import`.

There were also differences in the export syntax between each.

In terms of actual usage, there's a split between Node and browsers (and between older and newer versions of both). Modern browsers and modern Node can both handle `import`s natively. Older Node versions use `require` natively (but you can still write your code with `import`s and target old Node by transpiling to `require` syntax; technically what we do here because Google Cloud Functions run on Node 8 still...).

Browsers are even more complicated. Bundling JS for websites is a huge complicated topic, and is mostly due to the lack of widespread browser-native module support (although Webpack and other bundlers aren't going anywhere, they're also powerful tools for optimizations and other plugins).

Reply:
Short of it is that `import`s are the official winner. We'll eventually be able to use them everywhere without transipling (hugely optimistic, old browsers never go away...).